test/ExportVerilog/sv-dialect.mlir

```diff
@@ -133,10 +133,18 @@ hw.module @M1(%clock : i1, %cond : i1, %val : i8) {
 
       // CHECK-NEXT:     assert(cond);
       sv.assert %cond : i1
+      // CHECK-NEXT:     assert_0: assert(cond);
+      sv.assert {label = "assert_0"} %cond : i1
```

Comment:
Do you expect the label to be the normal case?  If so, it would probably be worth sugaring it a bit more, e.g. something like: `sv.assert "assert_0" %cond : i1`.  If they are unusual, then keeping them in the dict syntax makes sense.

Reply:
It's probably the normal case.

Do you know of a good way to do this without a custom printer/parser? This is trivial in ODS with `OptionalAttr<StrAttr>`, but I think needs a custom printer/parser for `StrAttr` where absence is interpreted as `""`.

Reply:
I think `custom<OmitEmtpyStringAttr>(label)` would do it?

Reply:
Good idea. Yes, that appears to work. The one annoyance is that `OpAsmParser` does not provide a `parseOptionalAttribute` method that does _not_ require a `NamedAttrList` parameter. This doesn't play well with `custom<...>` stuff which is going to do that loading for you after the fact. I'm using a dummy `NamedAttrList` to kludge around that. If I have time, I can see about an upstream fix. (Or maybe there's an approach I'm not seeing here. 🤔 )